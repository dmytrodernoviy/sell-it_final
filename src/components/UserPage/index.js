import React from 'react'
import UserPageStyle from './style';
import {connect} from 'react-redux'

const mapStateToProps = state => ({
    authorized: state.authorized
})

const UserPage = ({authorized}) => 
   <UserPageStyle>
        <div>
            <div className="user-photo" />
            <span>UserName: {authorized.userData.username || "Unknown"}</span>
            <span>Email: {authorized.userData.email || "Unknown"}</span>
            <span>First Name: {authorized.userData.first_name || "Unknown"}</span>
            <span>Last Name: {authorized.userData.last_name || "Unknown"}</span>
            <span>Location: {authorized.userData.location || "Unknown"}</span>
            <span>Color Scheme: {authorized.userData.color_scheme || "Unknown"}</span>
            <span>Language: {authorized.userData.language || "Unknown"}</span>
        </div>
    </UserPageStyle>

export default connect(mapStateToProps)(UserPage)