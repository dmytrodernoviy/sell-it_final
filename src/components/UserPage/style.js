import styled from 'styled-components'

const UserPageStyle = styled.div`
    background-color: #a39e9e;
    height: 100vh;
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    div {
        width: 40%;
        height: 90%;
        background: #00b3be;
        color: white;
        font-size: 22px;
        border-radius: 20px;
        padding-top: 20px;
        display: flex;
        flex-direction: column;
        justify-content: flex-start;
        align-items: center;
        span {
            margin-bottom: 15px;
        }
    }
    .user-photo {
        width: 170px;
        height: 170px;
        background: url(http://iephb.ru/wp-content/themes/iephb/images/default_user.png) no-repeat center;
        background-size: cover;
        border-radius: 30%;
        margin-bottom: 30px;
    }
    @media(max-width: 1024px) {
        div {
            width: 60%;
        }
    }
    @media(max-width: 768px) {
        div {
            width: 75%;
        }
    }
`
export default UserPageStyle