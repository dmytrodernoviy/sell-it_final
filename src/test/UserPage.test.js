import React from 'react';
import Enzyme, {mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import UserPage from '../components/UserPage';
import configureStore from 'redux-mock-store'
import { StaticRouter } from 'react-router'

Enzyme.configure({ adapter: new Adapter() });

describe('UserPage', () => {
    const initialState = {
        authorized: {
            userData: {
                username: "metal_master",
                email: "metalmaster@gmail.com",
                first_name: "",
                last_name: "",
                location: "",
                color_scheme: "",
                language: ""
            }
        }
    }
    const mockStore = configureStore()
    const store = mockStore(initialState)
    let Component;

    beforeEach(() => {
        Component = mount(
            <StaticRouter location="someLocation" context={{}}>
                <UserPage store={store}/>
            </StaticRouter>);
        })

    it("UserPage initial", () => {
        expect(Component.find('span')).toHaveLength(7)
        expect(Component.find('div')).toHaveLength(3)
        expect(Component.find('Link')).toHaveLength(1)
        expect(Component.find('button')).toHaveLength(1)
    })

    it("UserPage check values", () => {
        expect(Component.find('span').first().text()).toEqual('UserName: metal_master')
        expect(Component.find('span').at(1).text()).toEqual('Email: metalmaster@gmail.com')
        expect(Component.find('Link').prop('to')).toEqual('/')
        expect(Component.find('Button').prop('title')).toEqual('Return')
        expect(Component).toMatchSnapshot()
    })
  })