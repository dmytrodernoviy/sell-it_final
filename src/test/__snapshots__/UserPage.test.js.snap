// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPage UserPage check values 1`] = `ReactWrapper {}`;
